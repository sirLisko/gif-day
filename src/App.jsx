import React, { useState, useEffect } from 'react';
import styled from '@emotion/styled/macro';

import GifPicker from 'components/GifPicker';
import YearView from 'components/YearView';

const StyledHeader = styled.header`
  font-family: 'Luckiest Guy', cursive;
  text-align: center;
  margin: 3rem auto;
  h1 {
    font-size: 3rem;
  }
`;

const StyledContainer = styled.div`
  min-height: 50vh;
`;

const StyledFooter = styled.footer`
  text-align: center;
  margin-top: 2rem;
  a {
    color: currentColor;
    font-weight: bold;
  }
`;

const App = () => {
  const [dailyGifs, setDailyGifs] = useState({});
  const [selectedDay, setSelectedDay] = useState();

  useEffect(() => {
    const local = localStorage.getItem('dailyGifs');
    if (local) {
      try {
        const parsedLocal = JSON.parse(local);
        setDailyGifs(parsedLocal);
      } catch {
        localStorage.removeItem('dailyGifs');
      }
    }
  }, []);

  const saveGif = image => {
    const newdailyGifs = { ...dailyGifs, [selectedDay]: image };
    setDailyGifs(newdailyGifs);
    setSelectedDay();
    localStorage.setItem('dailyGifs', JSON.stringify(newdailyGifs));
  };

  return (
    <div>
      <StyledHeader>
        <h1>Have a gify day!</h1>
        <p>Your year in gifs...</p>
      </StyledHeader>
      {selectedDay && (
        <GifPicker
          selectedDay={selectedDay}
          selectedImg={dailyGifs[selectedDay]}
          onGifSelected={image => saveGif(image)}
        />
      )}

      <StyledContainer>
        <YearView dailyGifs={dailyGifs} onSelectedDay={setSelectedDay} />
      </StyledContainer>
      <StyledFooter>
        <p>
          Made with â™¥ by{' '}
          <a
            href="https://sirlisko.com"
            target="_blank"
            rel="noopener noreferrer"
          >
            Luca Lischetti (@sirLisko)
          </a>
        </p>
        <p>
          Powered By{' '}
          <a
            href="https://giphy.com/"
            target="_blank"
            rel="noopener noreferrer"
          >
            GIPHY
          </a>
        </p>
      </StyledFooter>
    </div>
  );
};

export default App;
