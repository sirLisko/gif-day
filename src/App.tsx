import React, { useState, useEffect } from "react";
import styled from "@emotion/styled/macro";

import GifPicker from "components/GifPicker";
import YearView from "components/YearView";
import { DailyGifs, Image } from "types";

const StyledHeader = styled.header`
  font-family: "Luckiest Guy", cursive;
  text-align: center;
  margin: 3rem auto;
  h1 {
    font-size: 3rem;
  }
`;

const StyledContainer = styled.div`
  min-height: 50vh;
`;

const StyledFooter = styled.footer`
  text-align: center;
  margin-top: 2rem;
  a {
    color: currentColor;
    font-weight: bold;
  }
`;

const App = () => {
  const [dailyGifs, setDailyGifs] = useState<DailyGifs>({});
  const [selectedDay, setSelectedDay] = useState<string>();

  useEffect(() => {
    const local = localStorage.getItem("dailyGifs");
    if (local) {
      try {
        const parsedLocal = JSON.parse(local);
        setDailyGifs(parsedLocal);
      } catch {
        localStorage.removeItem("dailyGifs");
      }
    }
  }, []);

  const saveGif = (image: Image) => {
    if (selectedDay) {
      const newdailyGifs = { ...dailyGifs, [selectedDay]: image };
      setDailyGifs(newdailyGifs);
      setSelectedDay(undefined);
      localStorage.setItem("dailyGifs", JSON.stringify(newdailyGifs));
    }
  };

  return (
    <div>
      <StyledHeader>
        <h1>Have a gify day!</h1>
        <p>Your year in gifs...</p>
      </StyledHeader>
      {selectedDay && (
        <GifPicker
          selectedDay={selectedDay}
          selectedImg={dailyGifs[selectedDay]}
          onClosePicker={(image) =>
            image ? saveGif(image) : setSelectedDay(undefined)
          }
        />
      )}

      <StyledContainer>
        <YearView dailyGifs={dailyGifs} onSelectedDay={setSelectedDay} />
      </StyledContainer>
      <StyledFooter>
        <p>
          Made with â™¥ by{" "}
          <a
            href="https://sirlisko.com"
            target="_blank"
            rel="noopener noreferrer"
          >
            Luca Lischetti (@sirLisko)
          </a>
        </p>
        <p>
          Powered By{" "}
          <a
            href="https://giphy.com/"
            target="_blank"
            rel="noopener noreferrer"
          >
            GIPHY
          </a>
        </p>
      </StyledFooter>
    </div>
  );
};

export default App;
