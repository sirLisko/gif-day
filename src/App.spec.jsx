import React from "react";
import { createRoot } from "react-dom/client";
import { render } from "@testing-library/react";

import App from "./App";
import { act } from "react-dom/test-utils";

it("renders without crashing", () => {
  const div = document.createElement("div");
  const root = createRoot(div);
  act(() => {
    root.render(<App />);
  });
  act(() => {
    root.unmount();
  });
});

describe("App Component", () => {
  let wrapper;

  beforeEach(async () => {
    wrapper = render(<App />);
  });

  it("should render properly", () => {
    expect(wrapper.getByRole("heading").innerHTML).toEqual("Have a gify day!");
  });
});
