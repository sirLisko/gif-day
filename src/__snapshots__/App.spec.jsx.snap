// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render properly 1`] = `
<div>
  <StyledHeader>
    <h1>
      Have a gify day!
    </h1>
    <p>
      Your year in gifs...
    </p>
  </StyledHeader>
  <StyledContainer>
    <YearView
      dailyGifs={Object {}}
      onSelectedDay={[Function]}
    />
  </StyledContainer>
  <StyledFooter>
    <p>
      Made with â™¥ by
       
      <a
        href="https://sirlisko.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        Luca Lischetti (@sirLisko)
      </a>
    </p>
    <p>
      Powered By
       
      <a
        href="https://giphy.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        GIPHY
      </a>
    </p>
  </StyledFooter>
</div>
`;
